api/src/main/java/net/kyori/adventure/pointer/Pointers.java

```diff
@@ -60,6 +65,36 @@
     return new PointersImpl.BuilderImpl();
   }
 
+  /**
+   * Gets all {@code pointer} in this collection which have the given value type.
+   *
+   * @param valueType the type class of the pointer
+   * @param <V> the type of the pointer
+   * @return an unmodifiable set of pointers filtered by the given type
+   * @since 4.10.0
+   */
+  @SuppressWarnings("unchecked")
+  default <V> @NotNull @Unmodifiable Set<Pointer<V>> pointers(final @NotNull Class<V> valueType) {
+    Objects.requireNonNull(valueType, "valueType");
+    final Set<Pointer<V>> filteredPointers = new HashSet<>();
+    for (final Pointer<?> pointer : this.pointers()) {
+      if (valueType.isAssignableFrom(pointer.type()))
+        filteredPointers.add((Pointer<V>) pointer);
+    }
+    return Collections.unmodifiableSet(filteredPointers);
+  }
+
+  /**
+   * Gets all {@code pointer} in this pointer collection.
+   *
+   * @return an unmodifiable set of pointers in this pointer collection
+   * @throws UnsupportedOperationException if the implementing class does not support querying for pointers
+   * @since 4.10.0
+   */
+  default @NotNull @Unmodifiable Set<Pointer<?>> pointers() {
+    throw new UnsupportedOperationException("The underlying Pointers implementation does not support querying for pointers.");
```

Comment:
Could we not theoretically make this default to `pointers(Object.class)` so we don't have to throw a UOE here?

Reply:
But that would be a endless loop as the `pointers(Class)` requires this method.
We could return an empty set but I thought that would be unpredictable functionality. And the Unsupported Operation only affects custom implementations.

Reply:
Ugh yeah... Man I wish we just made this class internal :')